File: ios/ConnectIgnoringSSLIOS.m
//
//  ConnectIgnoringSSLIOS
//  ZulipMobile
//
//  Created by Ebou Jobe on 13/04/2018.
//  Copyright © 2018 Facebook. All rights reserved.
//

// import RCTBridgeModule
#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include(“RCTBridgeModule.h”)
#import “RCTBridgeModule.h”
#else
#import “React/RCTBridgeModule.h” // Required when used as a Pod in a Swift project
#endif

// import RCTEventEmitter
#if __has_include(<React/RCTEventEmitter.h>)
#import <React/RCTEventEmitter.h>
#elif __has_include(“RCTEventEmitter.h”)
#import “RCTEventEmitter.h”
#else
#import “React/RCTEventEmitter.h” // Required when used as a Pod in a Swift project
#endif


@interface RCT_EXTERN_MODULE(ConnectIgnoringSSLIOS, NSObject)

RCT_EXTERN_METHOD(connect:(NSString *)url
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

@end

File: ios/ConnectIgnoringSSLIOS.swift
//
//  ConnectIgnoringSSLIOS
//  ZulipMobile
//
//  Created by Ebou Jobe on 13/04/2018.
//  Copyright © 2018 Facebook. All rights reserved.
//

import Foundation
import SwiftSocket

@objc(ConnectIgnoringSSLIOS)
class ConnectIgnoringSSLIOS : NSObject {
  // Export constants to use in your native module

  @objc func connect(_ url: String,
                     resolver resolve: RCTPromiseResolveBlock,
                     rejecter reject: RCTPromiseRejectBlock) -> Void {
    let client = TCPClient(address: url, port: 443)
    switch client.connect(timeout: 10) {
    case .success:
      resolve("successful")
      break
    case .failure(let error):
      let errorMsg = "Unable to connect to the url"
      let err: NSError = NSError(domain: errorMsg, code: 0, userInfo: nil)
      reject("Unsuccessful", errorMsg, err)
      break
    }
  }
}
